src/app/state/persist/saga.ts

```diff
@@ -0,0 +1,165 @@
+import { AnyAction } from '@reduxjs/toolkit'
+import { actionChannel, call, fork, put, select, take } from 'typed-redux-saga'
+import { isActionSynced } from 'redux-state-sync'
+import { persistActions, STORAGE_FIELD } from './index'
+import {
+  base64andStringify,
+  decryptWithPassword,
+  deriveKeyFromPassword,
+  encryptWithKey,
+  fromBase64andParse,
+} from './encryption'
+import { RootState } from 'types'
+import { EncryptedString, KeyWithSalt, PersistedRootState, SetUnlockedRootStatePayload } from './types'
+import { PasswordWrongError } from 'types/errors'
+import { walletActions } from 'app/state/wallet'
+import { selectUnlockedStatus } from 'app/state/selectUnlockedStatus'
+import { runtimeIs } from 'config'
+
+function* watchPersistAsync() {
+  yield* fork(function* () {
+    const channelQueue = yield* actionChannel<AnyAction>('*')
+    while (true) {
+      // TODO: could flush(channelQueue) to speedup multiple consecutive writes
+      const action: AnyAction = yield* take(channelQueue)
```

Comment:
non-security: does this result in every open tab trying to write on each change?